packages/relay/src/lib/eth.ts

```diff
@@ -482,30 +485,103 @@ export class EthImpl implements Eth {
-        : input;
+  private static prune0x(input: string): string {
+    return input.startsWith(EthImpl.emptyHex)
+      ? input.substring(2)
+      : input;
   }
 
   /**
    * Internal helper method that prepends a leading 0x if there isn't one.
    * @param input
    * @private
    */
-  private static prepend0x(input:string):string {
-    return input.startsWith('0x')
-        ? input
-        : '0x' + input;
+  private static prepend0x(input: string): string {
+    return input.startsWith(EthImpl.emptyHex)
+      ? input
+      : EthImpl.emptyHex + input;
   }
 
   /**
-   * Internal helper method that converts an ethAddress (with, or without a leading 0x)
-   * into an alias friendly AccountId.
-   * @param ethAddress
-   * @private
+   * Gets the block with the given hash.
+   * Given an ethereum transaction hash, call the mirror node to get the block info.
+   * Then using the block timerange get all contract results to get transaction details.
+   * If showDetails is set to true subsequently call mirror node for addtional transaction details
+   *
+   * TODO What do we return if we cannot find the block with that hash?
+   * @param blockHashOrNumber
+   * @param showDetails
    */
-  private static toAccountId(ethAddress:string) {
-    return AccountId.fromEvmAddress(0, 0, EthImpl.prune0x(ethAddress));
+  private async getBlock(blockHashOrNumber: number | string, showDetails: boolean): Promise<Block | null> {
+    const blockResponse = await this.mirrorNodeClient.getBlock(blockHashOrNumber);
+    if (blockResponse.hash === undefined) {
+      // block not found
+      return null;
+    }
+
+    const timestampRange = blockResponse.timestamp;
+    const timestampRangeParams = [`gte:${timestampRange.from}`, `lte:${timestampRange.to}`];
+    const contractResults = await this.mirrorNodeClient.getContractResults({ timestamp: timestampRangeParams });
+
+    // loop over contract function results to calculated aggregated datapoints
+    let gasUsed = 0;
+    let maxGasLimit = 0;
+    let timestamp = 0;
+    const transactions = [];
+    contractResults.results.forEach((result) => {
+      maxGasLimit = result.gas_limit > maxGasLimit ? result.gas_limit : maxGasLimit;
+      gasUsed += result.gas_used;
+      if (timestamp === 0) {
+        // The consensus timestamp of the first transaction in the block, with the nanoseconds part omitted.
+        timestamp = result.timestamp.substring(0, result.timestamp.indexOf('.')); // mirrorNode response assures format of ssssssssss.nnnnnnnnn
+      }
+    });
+
+    return new Block(null, null, {
+      baseFeePerGas: 0,
+      difficulty: EthImpl.zeroHex,
+      extraData: EthImpl.emptyHex,
+      gasLimit: maxGasLimit,
+      gasUsed: gasUsed,
+      hash: blockResponse.hash,
+      logsBloom: blockResponse.logsBloom,
+      miner: EthImpl.emptyHex,
+      mixHash: EthImpl.emptyHex,
+      nonce: EthImpl.emptyHex,
+      number: blockResponse.number,
+      parentHash: blockResponse.previous_hash,
```

Comment:
Needs to be trimmed to 256 bits
```suggestion
      parentHash: blockResponse.previous_hash.substring(0, 66),
```

Reply:
Fixed